### PR TITLE
feat: Add s3 upload output for capture

### PR DESCRIPTION
**Pull request by parkjeongryul**
# Description

This PR implement outputConfiguration.s3Upload feature.

## Related Issue

#201

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [x] I have followed the project's style guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

I've tested it working well in the two environments below.

### AWS s3
```
./bin/kubectl-retina  capture create --s3-region ap-northeast-2 --s3-bucket retina-test-bucket --s3-access-key-id my-access-key --s3-secret-access-key my-secret-key  --node-selectors "kubernetes.io/os=linux"  --debug
```

image

### minio
```
./bin/kubectl-retina  capture create --s3-endpoint https://play.min.io:9000  --s3-bucket test-jrpark-bucket --s3-access-key-id Q3AM3UQ867SPQQA43P2F --s3-secret-access-key zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG --node-selectors "kubernetes.io/os=linux"  --debug
```

image

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

**Comment by rbtr** [comment]
Hi @parkjeongryul, was something committed here that shouldn't have been? I don't see it anymore, but it was flagged

**Comment by parkjeongryul** [comment]
Hi @rbtr, I have made some modifications regarding the signing of commits and trivial indentation.
Could you please let me know if there are still any issues?

**Comment by rbtr** [comment]
Would you mind emailing retina@microsoft.com and mention this? Thanks!

**Comment by parkjeongryul** [comment]
@rbtr 
Could you help me understand which parts of this pull request are currently problematic? I don't think I'm understanding it very well. Could you please explain in more detail?

**Comment by rbtr** [comment]
@parkjeongryul I don't think there's anything problematic in this change right now. But a security issue was flagged when it was opened, I can give you the details in email?

**Comment by rbtr** [comment]
Sent the email to your address in your profile

**Comment by parkjeongryul** [comment]
> Sent the email to your address in your profile

I've replied, but sending to retina@microsoft.com seems to be blocked.

image

**Comment by parkjeongryul** [comment]
I will address the CI lint issues shortly and update the PR accordingly. Please hold off on further review until updated. Thanks!

**Comment by spencermckee** [comment]
@parkjeongryul Looks good to me. Thanks for the contribution. I'll approve it once the build is succeeding.

**Comment by parkjeongryul** [comment]
@spencermckee  I think that the CI lint issues have been resolved.

File: cli/cmd/capture/create.go
Comment (timraymond) [inline cli/cmd/capture/create.go:317]: Could you wrap this error?

File: pkg/capture/outputlocation/s3.go
Comment (timraymond) [inline pkg/capture/outputlocation/s3.go:162]: I'm curious about why there's platform-specific behavior here.

**Review by timraymond — COMMENTED** [review]
Looks pretty good. I just had a question and a small request.

**Review by spencermckee — DISMISSED** [review]

File: pkg/capture/outputlocation/s3.go
Comment (parkjeongryul) [reply pkg/capture/outputlocation/s3.go:162]: I implemented the code by referring to outputlocation/blob.go. I haven’t tested this on Windows. But I think we might need to set the secret path using the `CONTAINER_SANDBOX_MOUNT_POINT` env, considering that the absolute paths for volume mounts can differ in Windows.

https://kubernetes.io/docs/tasks/configure-pod-container/create-hostprocess-pod/#volume-mounts

https://github.com/microsoft/retina/blob/44a2a59c9d2f6dc02d578c503dc44cffe5eba2f9/pkg/capture/outputlocation/blob.go#L89-L100

File: cli/cmd/capture/create.go
Comment (parkjeongryul) [reply cli/cmd/capture/create.go]: done.

File: cli/cmd/capture/create.go
Comment (timraymond) [inline cli/cmd/capture/create.go:317]: Whoops, looks like the err was dropped
```suggestion
			return nil, fmt.Errorf("failed to create s3 upload secret: %w", err)
```

Comment (parkjeongryul) [reply cli/cmd/capture/create.go]: Thanks for catching that. updated.

**Review by spencermckee — APPROVED** [review]
Looks good to me.

**Pull request merged**
